core/blockchain_ext.go

```diff
@@ -0,0 +1,23 @@
+// (c) 2025, Ava Labs, Inc.
+package core
+
+import "github.com/ava-labs/libevm/metrics"
+
+// getOrOverrideAsRegisteredCounter searches for a metric already registered
+// with`name`. If a metric is found and it is a Counter, it is returned. If a
+// metric is found and it is not a Counter, it is unregistered and replaced with
+// a new registered Counter. If no metric is found, a new Counter is constructed
+// and registered.
+//
+// This is necessary for a metric defined in libevm with the same name but a
+// different type to what we expect.
+func getOrOverrideAsRegisteredCounter(name string, r metrics.Registry) metrics.Counter {
+	if r == nil {
+		r = metrics.DefaultRegistry
+	}
+	if c, ok := r.GetOrRegister(name, metrics.NewCounter).(metrics.Counter); ok {
```

Comment:
nit this is a bit of a long line, we could split it as

```suggestion
	m := r.GetOrRegister(name, metrics.NewCounter)
	if c, ok := m.(metrics.Counter); ok {
```

Reply:
I've split it but with a type switch. The call to `Unregister()` always made me a little uncomfortable so now the `default` branch gives a natural place for an explanatory comment.